### PR TITLE
[Tests] Improve createami test to validate and use the created AMI

### PR DESCRIPTION
**Pull request by hgreebe**
### Description of changes
We extended the test to use the generated AMI for a cluster and run a job on it.

We also extended the dimensions to run it on every supported OS and removed scheduler dimension.

### Tests
* Ran on personal Jenkins
```
test-suites:
  createami:
    test_createami.py::test_build_image:
      dimensions:
        - regions: ["eu-west-3"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: {{ common.OSS_COMMERCIAL_X86 }}
          schedulers: ["slurm"]
```

### References
* Modified https://github.com/aws/aws-parallelcluster/pull/6048

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/createami/test_createami.py [hunk 1]
```diff
@@ -122,6 +126,13 @@ def test_build_image(
     # remarkable AMIs are not available for ARM and ubuntu2204, centos7 yet
     if os not in ["ubuntu2204", "centos7"]:
         base_ami = retrieve_latest_ami(region, os, ami_type="remarkable", architecture=architecture)
+    elif os in ["ubuntu2204"]:
+        base_ami = retrieve_latest_ami(
+            region,
+            os,
+            architecture=architecture,
+            additional_filters=[{"Name": "creation-date", "Values": ["2023-01-06T*"]}],
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/createami/test_createami.py]: Can you identify a newer image here?  This date is quite old

File: tests/integration-tests/tests/createami/test_createami.py [hunk 2]
```diff
@@ -329,7 +345,7 @@ def _test_export_logs(s3_bucket_factory, image, region):
         output_file = f"{tempdir}/testfile.tar.gz"
         bucket_prefix = "test_prefix"
         ret = image.export_logs(bucket=bucket_name, output_file=output_file, bucket_prefix=bucket_prefix)
-        assert_that(ret["path"]).contains(output_file)
+        assert_that(ret["path"]).is_equal_to(output_file)
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/createami/test_createami.py]: why the stricter case here?

File: tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml [hunk 3]
```diff
@@ -23,3 +23,6 @@ DeploymentSettings:
         - {{ private_subnet_id }}
         SecurityGroupIds:
         - {{ default_vpc_security_group_id }}
+
+DevSettings:
+    DisableValidateAndTest: False
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: Is this necessary?

Comment (hgreebe) [reply tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: setting this parameter to false make it execute all the Kitchen tests with tag:install_ prefix as part of the Image Builder validation phase

Comment (dreambeyondorange) [reply tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: can this just be the default value?

**Review by dreambeyondorange — APPROVED** [review]

**Pull request merged**